llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -19716,10 +19716,15 @@ SDValue RISCVTargetLowering::PerformDAGCombine(SDNode *N,
     SDValue Src = N->getOperand(0);
     SDValue Passthru = N->getOperand(2);
     SDValue VL = N->getOperand(4);
-    // TODO: Handle fmv.v.f?
-    if (Src.getOpcode() == RISCVISD::VMV_V_X_VL && Passthru.isUndef() &&
-        VL == Src.getOperand(2))
-      return Src;
+    switch (Src.getOpcode()) {
+    default:
+      break;
+    case RISCVISD::VMV_V_X_VL:
+    case RISCVISD::VFMV_V_F_VL:
+      if (Passthru.isUndef() && VL == Src.getOperand(2))
+        return Src;
```

Comment:
I think we could also allow vmv.v.xs with mismatched VLs if their own passthru is undef, e.g.
```suggestion
      if (Passthru.isUndef() && (VL == Src.getOperand(2) || Src.getOperand(0).isUndef()))
        return Src;
```

Reply:
We can definitely generalize this in a couple dimensions.  I'm trying to make sure I have test coverage for everything I add, and haven't seen your suggested variant.  Feel free to check in a test case if you have one.